### PR TITLE
Add feature flags loading 

### PR DESCRIPTION
**Pull request by hackerstanislav**
JIRA: NAS-3638

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/api-client-tiger/src/profile.ts [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+export interface ILiveFeatures {
+    live: {
+        configuration: {
+            host: string;
+            key: string;
+        };
+        context: FeatureContext;
+    };
+}
+export interface IStaticFeatures {
```
Comment (no23reason) [inline libs/api-client-tiger/src/profile.ts:17]: These are not in any OpenAPI spec?

File: libs/sdk-backend-tiger/src/backend/features/index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import { getFeatureHubFeatures } from "./hub";
+import { getStaticFeatures } from "./static";
+
+export class TigerFeaturesService {
+    constructor(private readonly authCall: TigerAuthenticatedCallGuard) {}
+
+    public async getFeatures(profile: IUserProfile): Promise<ITigerFeatureFlags> {
+        return this.authCall(async (client) => {
+            const prof = profile || (await client.profile.getCurrent());
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/backend/features/index.ts:14]: If `profile` can be undefined, it should be marked as such in the `getFeatures` signature.

File: libs/sdk-backend-tiger/src/backend/features/hub.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    features: ILiveFeatures["live"],
+): Promise<Partial<ITigerFeatureFlags>> {
+    const { configuration, context } = features;
+    try {
+        const data = await loadHubFeatures(configuration, state);
+        const featuresMap = data.reduce((prev, item) => ({ ...prev, [item.key]: item }), {} as FeaturesMap);
+
+        return mapFeatures(featuresMap, context);
+    } catch (_err) {
+        return {};
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/backend/features/hub.ts:28]: Should we warn or err in the console at least?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8277/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10128/ : SUCCESS in 13m 40s

File: libs/api-client-tiger/src/profile.ts (hunk 1 above)
Comment (hackerstanislav) [reply libs/api-client-tiger/src/profile.ts:17]: No :(

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8279/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10131/ : SUCCESS in 13m 01s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2375/ : SUCCESS in 1m 16s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2669/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1487/ : SUCCESS in 33m 02s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10133/ : SUCCESS in 15m 33s

**Pull request merged**